core/blockchain.go

```diff
@@ -413,7 +413,10 @@ func NewBlockChain(
 // unindexBlocks unindexes transactions depending on user configuration
 func (bc *BlockChain) unindexBlocks(tail uint64, head uint64, done chan struct{}) {
 	start := time.Now()
-	txLookupLimit := bc.cacheConfig.TxLookupLimit
+	if bc.cacheConfig.TxLookupLimit < 0 {
```

Comment:
Because `unindexBlocks` is spawned by `dispatchTxUnindexer`, do we need to worry about this?

Reply:
yup correct. I put it to be extra safe (in case this is directly called by tests), but I guess it's fine to just add it as a comment.